packages/amazon-bedrock/src/bedrock-chat-language-model.ts

```diff
@@ -377,19 +387,111 @@ export class BedrockChatLanguageModel implements LanguageModelV1 {
               }
             }
           },
-
           flush(controller) {
             controller.enqueue({
               type: 'finish',
               finishReason,
               usage,
-              providerMetadata,
+              ...(providerMetadata && { providerMetadata }),
             });
           },
         }),
       ),
       rawCall: { rawPrompt, rawSettings },
+      rawResponse: { headers: responseHeaders },
       warnings,
     };
   }
+
+  private getUrl(modelId: string) {
+    const encodedModelId = encodeURIComponent(modelId);
+    return `${this.config.baseUrl()}/model/${encodedModelId}/converse`;
+  }
+
+  private getStreamUrl(modelId: string): string {
+    const encodedModelId = encodeURIComponent(modelId);
+    return `${this.config.baseUrl()}/model/${encodedModelId}/converse-stream`;
+  }
```

Comment:
Alternative: have a `getUrl` method and then use `${this.getUrl(modelId)/converse` and `${this.getUrl(modelId)/converse-stream` - either way works